tests/test/util/test_environment.cpp

```diff
@@ -35,4 +37,16 @@ TEST_CASE("Test setting environment variables", "[util]")
     const std::string original2 = setEnvVar("MY_VAR", "gamma");
     REQUIRE(original2 == "beta");
 }
+
+TEST_CASE("Test overriding CPU count", "[util]")
+{
+    // Unset just in case it is set
+    unsetEnvVar("OVERRIDE_CPU_COUNT");
+
+    unsigned int expectedNumCores = std::thread::hardware_concurrency();
+    REQUIRE(getUsableCores() == expectedNumCores);
+
+    getSystemConfig().overrideCpuCount = 4;
+    REQUIRE(getUsableCores() == 4);
```

Comment:
This again will leave the modifications hanging around for any tests that don't call `cleanFaabric`, before running. I'd recommend this instead:

```
cleanFaabric();

// Check default cores is same as available concurrency
auto &conf = getSystemConfig();
int defaultCores = getUsableCores();
REQUIRE(defaultCores == std::thread::hardware_concurrency());

// Check it can be overridden (using a number that's obviously always wrong)
conf.overrideCpuCount = 1234;
REQUIRE(getUsableCores() == 1234);

// Reset the conf
conf.reset();

// Check we're back
REQUIRE(getUsableCores() == defaultCores);
```